### PR TITLE
Feat: Add support for the require_last_push_approval flag in github_branch_protection_v3

### LINKED ISSUE #1723: [FEAT]: Support require_last_push_approval in github_branch_protection_v3
### Describe the need

We use github_branch_protection_v3 to manage branch protection rules across hundreds of repos, but it lacks the [require_last_push_approval](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/branch_protection#require_last_push_approval) argument. This control is necessary to prevent [Pull Request Hijacking](https://www.legitsecurity.com/blog/bypassing-github-required-reviewers-to-submit-malicious-code), so please make it possible to enable via `github_branch_protection_v3`.

### SDK Version

_No response_

### API Version

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by matthewcostatide**
Resolves #1723 

----

### Before the change?
The `github_branch_protection_v3` resource doesn't support setting the `require_last_push_approval` flag, and its presence is omitted from the documentation. 

### After the change?
This PR adds support for the `require_last_push_approval` flag in the `required_pull_request_reviews` block  of the `github_branch_protection_v3` resource type, consistent with the `github_branch_protection` resource.

### Pull request checklist
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?
It should not

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes
- [x] No

----

### CODE REVIEW

File: github/resource_github_branch_protection_v3.go [hunk 1]
```diff
@@ -128,6 +128,12 @@ func resourceGithubBranchProtectionV3() *schema.Resource {
 							Description:  "Require 'x' number of approvals to satisfy branch protection requirements. If this is specified it must be a number between 0-6.",
 							ValidateFunc: validation.IntBetween(0, 6),
 						},
+						"require_last_push_approval": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Default:     false,
+							Description: "Require that The most recent push must be approved by someone other than the last pusher.",
```
Comment (kfcampbell) [inline github/resource_github_branch_protection_v3.go:135]:
```suggestion
							Description: "Require that the most recent push must be approved by someone other than the last pusher.",
```

File: website/docs/r/branch_protection_v3.html.markdown [hunk 2]
```diff
@@ -112,6 +112,7 @@ The following arguments are supported:
 * `require_code_owner_reviews`: (Optional) Require an approved review in pull requests including files with a designated code owner. Defaults to `false`.
 * `required_approving_review_count`: (Optional) Require x number of approvals to satisfy branch protection requirements. If this is specified it must be a number between 0-6. This requirement matches GitHub's API, see the upstream [documentation](https://developer.github.com/v3/repos/branches/#parameters-1) for more information.
 * `bypass_pull_request_allowances`: (Optional) Allow specific users, teams, or apps to bypass pull request requirements. See [Bypass Pull Request Allowances](#bypass-pull-request-allowances) below for details.
+* `require_last_push_approval`: (Optional) Require that The most recent push must be approved by someone other than the last pusher.  Defaults to `false`
```
Comment (kfcampbell) [inline website/docs/r/branch_protection_v3.html.markdown:115]:
```suggestion
* `require_last_push_approval`: (Optional) Require that the most recent push must be approved by someone other than the last pusher.  Defaults to `false`
```

File: github/resource_github_branch_protection_v3_utils.go [hunk 3]
```diff
@@ -351,6 +351,7 @@ func expandRequiredPullRequestReviews(d *schema.ResourceData) (*github.PullReque
 			rprr.DismissStaleReviews = m["dismiss_stale_reviews"].(bool)
 			rprr.RequireCodeOwnerReviews = m["require_code_owner_reviews"].(bool)
 			rprr.RequiredApprovingReviewCount = m["required_approving_review_count"].(int)
+			*rprr.RequireLastPushApproval = m["require_last_push_approval"].(bool)
```
Comment (kfcampbell) [inline github/resource_github_branch_protection_v3_utils.go:354]: Do you mind explaining the difference between the code you've written and doing something like:

```go
            requireLastPushApproval := m["require_last_push_approval"].(bool)
// ...
			rprr.RequireLastPushApproval = &requireLastPushApproval
```

Comment (matthewcostatide) [reply github/resource_github_branch_protection_v3_utils.go]: The difference is my code will NPE if rprr.RequireLastPushApproval is null, which it is likely to be unless that field is explicitly constructed with the rest of the struct.

I guess the bigger, and slightly out of scope for this PR, question is why is that field a pointer?

**Comment by matthewcostatide** [comment]
@kfcampbell any chance you can take a look at this?

**Comment by georgekaz** [comment]
It would be nice if someone would look at this. Is there any reason it's sitting around being ignored?

**Comment by kfcampbell** [comment]
Sorry for the delay here. After the addition in `TestAccGithubBranchProtectionV3_required_pull_request_reviews`, the acceptance tests are failing: 

```
    testing.go:705: Step 0 error: Check failed: 1 error occurred:
        	* Check 5/12 error: github_branch_protection_v3.test: Attribute 'required_pull_request_reviews.0.require_last_push_approval' expected "true", got "false"
```

@matthewcostatide can you reproduce this? Do the tests succeed for you?

**Comment by georgekaz** [comment]
@kfcampbell @matthewcostatide I had a look at this and I think what is missing is 
`"require_last_push_approval":      rprr.RequireLastPushApproval,` 
from this block: https://github.com/matthewcostatide/terraform-provider-github/blob/branch_protection_v3_require_last_push_approval/github/resource_github_branch_protection_v3_utils.go#L191-L201

```
		return d.Set("required_pull_request_reviews", []interface{}{
			map[string]interface{}{
				"dismiss_stale_reviews":           rprr.DismissStaleReviews,
				"dismissal_users":                 schema.NewSet(schema.HashString, users),
				"dismissal_teams":                 schema.NewSet(schema.HashString, teams),
				"dismissal_apps":                  schema.NewSet(schema.HashString, apps),
				"require_code_owner_reviews":      rprr.RequireCodeOwnerReviews,
				"require_last_push_approval":      rprr.RequireLastPushApproval,
				"required_approving_review_count": rprr.RequiredApprovingReviewCount,
				"bypass_pull_request_allowances":  bpra,
			},
		})
```

The tests seem pretty broken to me, they fail to delete the repos they create with a `403 Must have admin rights to Repository. []` error. However, prior to my suggested change I saw the error that @kfcampbell mentions above and after adding this line, that error is resolved.

On another note, this bit of code
```
requireLastPushApproval := m["require_last_push_approval"].(bool)
rprr.RequireLastPushApproval = &requireLastPushApproval
```
as opposed to 
```
rprr.DismissStaleReviews = m["dismiss_stale_reviews"].(bool)
```
seems to be required because of a strange use of a pointer [here](https://github.com/matthewcostatide/terraform-provider-github/blob/branch_protection_v3_require_last_push_approval/vendor/github.com/google/go-github/v55/github/repos.go#L1118) and I think [this accessor](https://github.com/matthewcostatide/terraform-provider-github/blob/branch_protection_v3_require_last_push_approval/vendor/github.com/google/go-github/v55/github/github-accessors.go#L16361-L16367) code is possibly wrong too. Something to look at outside the scope of this PR.

**Comment by kfcampbell** [comment]
@georgekaz you're a hero for fighting the good fight with our test situation. Thanks for your research! As always, should you desire to submit any PRs with fixes you'd like to see, I'm happy to review.

**Comment by georgekaz** [comment]
Yeah, sure I can take a look at the tests in general. On this occasion though the test was correct to fail, the property on the object is not being set. If @matthewcostatide would like to add the fix I suggested above it will pass. I don't really want to push to Matthew's branch, he may not like it!

**Comment by georgekaz** [comment]
Hi @matthewcostatide could you add the following line please:

`"require_last_push_approval":      rprr.RequireLastPushApproval,`

here: https://github.com/matthewcostatide/terraform-provider-github/blob/branch_protection_v3_require_last_push_approval/github/resource_github_branch_protection_v3_utils.go#L197C6-L197C32

Then I think the tests will pass and this PR has a chance of being merged. Thanks

**Comment by aalvarezaph** [comment]
>  I don't really want to push to Matthew's branch, he may not like it!

IMO given this was raised beginning on Feb, seems fair enough to push this fix to the branch . This is a feature many will be able to benefit from :)

**Comment by georgekaz** [comment]
@aalvarezaph @kfcampbell I couldn't edit @matthewcostatide's branch because I'm not a repo maintainer, so I've had to open a new PR with the patch. https://github.com/integrations/terraform-provider-github/pull/2199

**Pull request closed without merging**